memcache/memcache.go

```diff
@@ -711,3 +739,238 @@ func (c *Client) incrDecr(verb, key string, delta uint64) (uint64, error) {
+	return
+}
+
+func (c *Client) onMetaItem(item *MetaSetItem, fn func(*Client, *bufio.ReadWriter, *MetaSetItem, func(metaData *MetaResponseMetadata)) error, cb func(metaData *MetaResponseMetadata)) error {
+	addr, err := c.selector.PickServer(item.Key)
+	if err != nil {
+		return err
+	}
+	cn, err := c.getConn(addr)
+	if err != nil {
+		return err
+	}
+	defer cn.condRelease(&err)
+	if err = fn(c, cn.rw, item, cb); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Client) processMetaSet(rw *bufio.ReadWriter, item *MetaSetItem, cb func(*MetaResponseMetadata)) error {
+	if !legalKey(item.Key) {
+		return ErrMalformedKey
+	}
+
+	command := c.parseFlagsForMetaSet(item)
+
+	var err error
+	_, err = fmt.Fprintf(rw, command)
+
+	if err != nil {
+		return nil
+	}
+	if _, err = rw.Write(item.Value); err != nil {
+		return err
+	}
+	if _, err := rw.Write(crlf); err != nil {
+		return err
+	}
+	if err := rw.Flush(); err != nil {
+		return err
+	}
+	response, err := rw.ReadSlice('\n')
+	if err != nil {
+		return err
+	}
+	switch {
+	case bytes.HasPrefix(response, metaResultStored):
+		metaResponseMetadata := new(MetaResponseMetadata)
+		if err = parseMetaResponseMetadata(response, metaResponseMetadata); err != nil {
+			return err
+		}
+		cb(metaResponseMetadata)
+		return nil
+	case bytes.HasPrefix(response, metaResultNotStored):
+		return ErrNotStored
+	case bytes.HasPrefix(response, metaResultExists):
+		return ErrCASConflict
+	case bytes.HasPrefix(response, metaResultNotFound):
+		return ErrCacheMiss
+	}
+	return fmt.Errorf("memcache: unexpected response line from ms: %q", string(response))
+}
+
+func getMemCacheErrorFromResponse(response []byte) error {
+	switch {
+	case bytes.HasPrefix(response, resultClientErrorPrefix):
+		errMsg := response[len(resultClientErrorPrefix) : len(response)-2]
+		return errors.New("memcache: client error: " + string(errMsg))
+
+	case bytes.HasPrefix(response, resultServerErrorPrefix):
+		errMsg := response[len(resultServerErrorPrefix) : len(response)-2]
+		return errors.New("memcache: server error: " + string(errMsg))
+	}
+
+	return nil
+}
+
+//function reads the  response from meta commands and returns a struct MetaResponseMetadata
+//response of meta commands contain flags which are single characters.
+// For ex if MetaGetFlags.ReturnTTLRemainingSecondsInResponse is set to true then response
+//will contain t300. Here 300 is the amount of TTL remaining in Seconds
+func parseMetaResponseMetadata(response []byte, respMetadata *MetaResponseMetadata) error {
+	// the first two characters are being processed in this switch-case block, the other cases are all errors
+	// so we don't need to save them into the meta data object
+	metaResponseMetadata := strings.Fields(string(response))[1:]
+	if len(metaResponseMetadata) != 0 {
+		for _, metadata := range metaResponseMetadata {
+			if err := populateMetaResponseMetadata(metadata[0:1], metadata[1:], respMetadata); err != nil {
```

Comment:
Lets modify populateMetaResponseMetadata to accept just string and then it can decide to split it into key and value. I believe this was your recommendation. I have made the same implementation [here](https://github.com/spaiStrp/gomemcache/blob/b1b62dcca2753cd73b759f2d7dda04a7e08b0bba/memcache/memcache.go#L731). Let me know what you think.